design/sales.md

```diff
@@ -140,6 +140,47 @@ the state is kept on local disk by the Repo and the Datastore. How much space is
 reserved to be sold is persisted on disk by the Repo. The availabilities are
 persisted on disk by the Datastore.
 
+Request queue
+----------
+Once a new request for storage is created on chain, all hosts will receive a
+contract event announcing the storage request and decide if they want to act on
+the request by matching their availabilities with the incoming request. Because
+there will be many requests being announced over time, each host will create a
+queue of matching requests, adding each new storage request to the queue.
+Requests in the queue will be sorted in the following order:
+1. Profit (descending, yet to be determined how this will be calculated)
+2. Collateral required (ascending)
+3. Time before expiry (descending)
+4. Data size (ascending)
```

Comment:
Hmm, what role does data size play in decision-making? I understand the roles of the previous three points, but I am unsure about this one. Smaller requests are better? Why? Maybe because it allows the host to more efficiently cover its availabilities? But then bigger requests will be "penalized" and might take more time to fulfill... Is that desired?

Reply:
This is a good point. Larger data sets will cost more to download in both cost and time. It stands to reason that eventually dataset size would have an impact on profitability, as downloading the data will incur a cost in bandwidth fees. But as a first implementation attempt, profit will not be included as a sorting parameter (as we don't know how to calculate it yet) and dataset size could temporarily fill its shoes.

Yes, it seems the implications of considering dataset size in sorting the requests could consequently cause larger dataset requests to take more time to fill, when comparing to requests assuming that all other parameters of the request are the same. However, hopefully this is resolved once profitability can be calculated with dataset size included in that calculation.

Reply:
I've added a note regarding this in the `Sort order` section.